portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,115 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var context;
+var canvas;
+var myBubbleArray;
+var numBubs = 20;
+var animationID = null;
+// Resizes bubbles when page size is changed
+window.addEventListener('resize', init)
+
+function startChat() {
+  grabComments();
+  init();
+  grabUser();
+}
+
+class Bubble {
+  constructor(color) {
+    this.speed_factor = 3;
+    if (canvas.width < 900) {
+      this.speed_factor = 1.75;
+    }
+    this.color = color;
+    this.size = Math.random() * canvas.width;
+    this.x = Math.random() * canvas.width;
+    this.y = Math.random() * canvas.height;
+    this.radius = Math.floor(Math.random() * Math.sqrt(canvas.width));
+    this.dx = Math.random() * this.speed_factor * this.randomDirection();
+    this.dy = Math.random() * this.speed_factor * this.randomDirection();
+    this.opacity = Math.random();
+  }
+  drawBubble() {
+    context.beginPath();
+    context.fillStyle = this.color;
+    context.globalAlpha = this.opacity;
+    context.arc(this.x, this.y, this.radius, 0, Math.PI * 2, true);
+    context.closePath();
+    context.fill();
+  }
+  randomDirection() {
```

Comment:
Since this doesn't interact with Bubble's state, you could make this a free function (a function outside of a class).

Reply:
Ping